### PR TITLE
[feature] Modify function behavior

### PR DESCRIPTION
**Pull request by abought**
Per discussion w/ @sloria . To help review and choose, email with context to follow.

### CODE REVIEW

**Comment by abought** [comment]
Force-restarted failing tests; clear now.

**Comment by lyndsysimon** [comment]
This seems like a bad idea - why are we unescaping only a subset of characters?

**Comment by lyndsysimon** [comment]
We spoke about this in person - I understand the need for it, I'm concerned about confusion about the function's purpose and when it should be used.

Please change the name of `safe_unescape_html` to `unencode_ampersands` or similar. Also, revise the function's docstring to be more concise on the first line (e.g.: "Converts HTML-encoded ampersands to literal characters") and document the rationale for the function's existence in the next section.

I'll also add this function to the list of things to clean up after this refactor and migration are complete.

**Comment by lyndsysimon** [comment]
The above documentation changes can be made in the related PR, #3659.

@sloria, please proceed :)

**Comment by abought** [comment]
This function is used primarily in views tied to `json_renderer`. It is designed to turn HTML entities in the database back to regular text, under the expectation that the text is being passed to a front-end framework that does its own escaping. (eg knockout)

**Comment by abought** [comment]
Proposed name: `unescape_entities`. (it may still handle things other than ampersands if needed)

I'd suggest this remain separate from any of the markupsafe conversion PRs- I'd rather keep those focused on mako. 

Will push requested changes to this PR branch momentarily.

File: website/mails.py [hunk 1]
```diff
@@ -180,7 +180,7 @@ def send_mail(to_addr, mail, mimetype='plain', from_addr=None, mailer=None,
     subject='Your ${action} has failed',
 )
 
-UNESCAPE = "<% from website.util.sanitize import safe_unescape_html %> ${safe_unescape_html(src.title)}"
+UNESCAPE = "<% from website.util.sanitize import unescape_entities %> ${unescape_entities(src.title)}"
```
Comment (abought) [inline website/mails.py]: It seems a bit weird to me that a python file has embedded Mako logic in a string; possible separation of concerns issue? Probably outside scope for this PR though.

**Comment by abought** [comment]
Updated with newest develop. Will write test cases Monday AM.

**Comment by sloria** [comment]
Thanks. Once those tests are finished, please ping me on Flowdock for next pass.

**Comment by sloria** [comment]
Merged into `feature/markupsafe-3`.

**Pull request closed without merging**